flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServerConnection.java

```diff
@@ -235,8 +235,54 @@ else if (contentAddressable == CONTENT_ADDRESSABLE) {
 			return;
 		}
 
-		// from here on, we started sending data, so all we can do is close the connection when something happens
+		readLock.lock();
+
 		try {
+			try {
+				if (!blobFile.exists()) {
+					// first we have to release the read lock in order to acquire the write lock
+					readLock.unlock();
+					writeLock.lock();
```

Comment:
True, I'm also not sure which version would be more efficient. Given that the file transfer dominates the execution time here, I assume that the additional check won't hurt.